burr/core/persistence.py

```diff
@@ -192,6 +195,17 @@ def initialize(self):
         """Creates the table if it doesn't exist"""
         # Usage
         self.create_table_if_not_exists(self.table_name)
+        self._initialized = True
+
+    def is_initialized(self) -> bool:
+        """Check if persister is properly initialized."""
+        if self._initialized:
+            return True
+        cursor = self.connection.cursor()
+        cursor.execute(
+            "SELECT name FROM sqlite_master WHERE type='table' AND name=?", (self.table_name,)
+        )
+        return cursor.fetchone() is not None
```

Comment:
You could also set `self._initialized` to `True` here -- then maybe reuse the same one? E.G. cache it? It'll never go from initialized to uninitialized...

Reply:
oh sure, yeah we should do that.